src/main/scala/dpla/ingestion3/reports/PropertyDistinctValueReport.scala

```diff
@@ -77,32 +79,82 @@ class PropertyDistinctValueReport(
+        ds.map(dplaMapData => PropertyDistinctValueRpt(value = extractValue(dplaMapData.sourceResource.format)))
+      case "sourceResource.genre" =>
+        ds.map(dplaMapData => PropertyDistinctValueRpt(extractValue(dplaMapData.sourceResource.genre)))
+      case "sourceResource.identifier" =>
+        ds.map(dplaMapData => PropertyDistinctValueRpt(extractValue(dplaMapData.sourceResource.identifier)))
+      case "sourceResource.language.providedLabel" =>
+        ds.map(dplaMapData => PropertyDistinctValueRpt(extractValue(dplaMapData.sourceResource.language)))
+      case "sourceResource.place.name" =>
+        ds.map(dplaMapData => PropertyDistinctValueRpt(extractValue(dplaMapData.sourceResource.place)))
+      case "sourceResource.publisher.name" =>
+        ds.map(dplaMapData => PropertyDistinctValueRpt(extractValue(dplaMapData.sourceResource.publisher)))
+      case "sourceResource.relation" =>
+        ds.map(dplaMapData => PropertyDistinctValueRpt(extractValue(dplaMapData.sourceResource.relation)))
+      case "sourceResource.replacedBy" =>
+        ds.map(dplaMapData => PropertyDistinctValueRpt(extractValue(dplaMapData.sourceResource.replacedBy)))
+      case "sourceResource.replaces" =>
+        ds.map(dplaMapData => PropertyDistinctValueRpt(extractValue(dplaMapData.sourceResource.replaces)))
       case "sourceResource.rights" =>
-        ds.map(dplaMapData => dplaMapData.sourceResource.rights)
-          .flatMap(x => x)
-          .groupBy("value")
-          .count
+        ds.map(dplaMapData => PropertyDistinctValueRpt(extractValue(dplaMapData.sourceResource.rights)))
+      case "sourceResource.rightsHolder" =>
+        ds.map(dplaMapData => PropertyDistinctValueRpt(extractValue(dplaMapData.sourceResource.rightsHolder)))
+      case "sourceResource.subject.providedLabel" =>
+        ds.map(dplaMapData => PropertyDistinctValueRpt(extractValue(dplaMapData.sourceResource.subject)))
+      case "sourceResource.temporal.originalSourceDate" =>
+        ds.map(dplaMapData => PropertyDistinctValueRpt(extractValue(dplaMapData.sourceResource.temporal)))
+      case "sourceResource.title" =>
+        ds.map(dplaMapData => PropertyDistinctValueRpt(extractValue(dplaMapData.sourceResource.title)))
       case "sourceResource.type" =>
-        ds.map(dplaMapData => dplaMapData.sourceResource.`type`)
-          .flatMap(x => x)
-          .groupBy("value")
-          .count
+        ds.map(dplaMapData => PropertyDistinctValueRpt(extractValue(dplaMapData.sourceResource.`type`)))
       case x =>
         throw new RuntimeException(s"Unrecognized field name '$x'")
     }
 
+    makeCsvTable(rptDs, spark, token)
   }
 
+
+  /**
+    * Explodes the value column so that multiple values
+    * occur on separate rows. Then the groupBy and count
+    * operations are performed on the data
+    *
+    * @param rptDataset
+    * @param spark
+    * @param token
+    * @return
+    */
+  def makeCsvTable(rptDataset: Dataset[PropertyDistinctValueRpt],
```

Comment:
This may a bit nit-picky, but would you consider calling this method just `makeTable` since it's not actually writing to a CSV file?  I was a bit confused at first.